nixos/modules/services/web-apps/rutorrent.nix

```diff
@@ -0,0 +1,281 @@
+          You are using exposeInsecureRPC2mount without using basic auth on the virtual host. The exposed rpc mount allow for remote command execution.
+
+          Please make sure it is not accessible from the outside.
+        '');
+
+      systemd = {
+        services = {
+          rtorrent.path = getPluginDependencies cfg.plugins;
+          rutorrent-setup = let
+            rutorrentConfig = pkgs.writeText "rutorrent-config.php" ''
+              <?php
+                // configuration parameters
+
+                // for snoopy client
+                @define('HTTP_USER_AGENT', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.87 Safari/537.36', true);
+                @define('HTTP_TIME_OUT', 30, true);	// in seconds
+                @define('HTTP_USE_GZIP', true, true);
+                $httpIP = null;				// IP string. Or null for any.
+                $httpProxy = array
+                (
+                  'use' 	=> false,
+                  'proto'	=> 'http',		// 'http' or 'https'
+                  'host'	=> 'PROXY_HOST_HERE',
+                  'port'	=> 3128
+                );
+
+                @define('RPC_TIME_OUT', 5, true);	// in seconds
+
+                @define('LOG_RPC_CALLS', false, true);
+                @define('LOG_RPC_FAULTS', true, true);
+
+                // for php
+                @define('PHP_USE_GZIP', false, true);
+                @define('PHP_GZIP_LEVEL', 2, true);
+
+                $schedule_rand = 10;			// rand for schedulers start, +0..X seconds
+
+                $do_diagnostic = true;
+                $log_file = '${cfg.home}/logs/errors.log';		// path to log file (comment or leave blank to disable logging)
+
+                $saveUploadedTorrents = true;		// Save uploaded torrents to profile/torrents directory or not
+                $overwriteUploadedTorrents = false;     // Overwrite existing uploaded torrents in profile/torrents directory or make unique name
+
+                $topDirectory = '/';			// Upper available directory. Absolute path with trail slash.
+                $forbidUserSettings = false;
+
+                $scgi_port = 0;
+                $scgi_host = "unix://${cfg.rpcSocket}";
+
+                $XMLRPCMountPoint = "/RPC2";		// DO NOT DELETE THIS LINE!!! DO NOT COMMENT THIS LINE!!!
+
+                $pathToExternals = array(
+                  "php" 	=> ''',			// Something like /usr/bin/php. If empty, will be found in PATH.
+                  "curl"	=> ''',			// Something like /usr/bin/curl. If empty, will be found in PATH.
+                  "gzip"	=> ''',			// Something like /usr/bin/gzip. If empty, will be found in PATH.
+                  "id"	=> ''',			// Something like /usr/bin/id. If empty, will be found in PATH.
+                  "stat"	=> ''',			// Something like /usr/bin/stat. If empty, will be found in PATH.
```

Comment:
Should we fill these to ensure that they're fetched as dependencies? Or are these dependencies optional and only required by certain plugins?

Reply:
See the last comment. I'll fill these.